selfdrive/ui/qt/maps/map_instructions.cc

```diff
@@ -102,6 +102,9 @@ void MapInstructions::updateInstructions(cereal::NavInstruction::Reader instruct
     icon_01->setVisible(true);
   }
 
+  // Hide distance after arrival
+  distance->setVisible(type != "arrive" || instruction.getManeuverDistance() > 0);
```

Comment:
Need both conditions, because distance can be positive when arriving as you approach and you still want to show distance:

<details>
<summary>View</summary>
( logMonoTime = 1218725162203429,
  valid = true,
  navInstruction = (
    maneuverPrimaryText = "You have arrived at your destination",
    maneuverDistance = 2.8782964,
    maneuverType = "arrive",
    maneuverModifier = "straight",
    distanceRemaining = 2.8782964,
    timeRemaining = 0.68005437,
    timeRemainingTypical = 0.68005437,
    showFull = true,
    speedLimit = 11.111111,
    speedLimitSign = mutcd ) )
</details>

And it can also be negative for normal instructions once you pass them for a short while:

<details>
<summary>View</summary>
( logMonoTime = 1218884832178237,
  valid = true,
  navInstruction = (
    maneuverPrimaryText = "Pacific Coast Highway",
    maneuverDistance = -12.218868,
    maneuverType = "turn",
    maneuverModifier = "right",
    distanceRemaining = 31.553131,
    timeRemaining = 8.2187948,
    timeRemainingTypical = 8.2187948,
    showFull = true,
    speedLimit = 11.111111,
    speedLimitSign = mutcd ) )
</details>